### PR TITLE
[Master] Fix volume not found

### PR DESCRIPTION
**Pull request by grolu**
**What this PR does / why we need it**:
Fixes as in Hotifx #503
In addition, this PR fixes unavailable machine types / volume types handling.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```improvement operator
Fixed: Unavailable machine types / volume types for specific zones were not handled correctly
```

### CODE REVIEW

**Comment by grolu** [comment]
Do not merge, until #511 is merged

File: frontend/src/components/ShootWorkers/WorkerConfiguration.vue [hunk 1]
```diff
@@ -83,7 +83,8 @@ export default {
 
       const workers = cloneDeep(this.shootWorkerGroups)
       const zonesNetworkConfiguration = get(this.shootItem, 'spec.provider.infrastructureConfig.networks.zones')
-      this.$refs.manageWorkers.setWorkersData({ workers, cloudProfileName: this.shootCloudProfileName, region: this.shootRegion, zonesNetworkConfiguration })
+      const zonedCluster = get(this.shootItem, 'spec.provider.infrastructureConfig.zoned', false)
```
Comment (petersutter) [inline frontend/src/components/ShootWorkers/WorkerConfiguration.vue]: check

File: frontend/src/components/ShootWorkers/WorkerGroup.vue [hunk 2]
```diff
@@ -54,15 +63,15 @@ export default {
...
   computed: {
     ...mapGetters([
-      'machineTypesByCloudProfileNameAndZones',
-      'volumeTypesByCloudProfileNameAndZones',
+      'machineTypesByCloudProfileName',
+      'volumeTypesByCloudProfileName',
       'machineImagesByCloudProfileName'
     ]),
     machineTypes () {
-      return this.machineTypesByCloudProfileNameAndZones({ cloudProfileName: this.cloudProfileName })
```
Comment (petersutter) [inline frontend/src/components/ShootWorkers/WorkerGroup.vue]: check machineTypesByCloudProfileNameAndZones function (zone param)

File: frontend/src/components/ShootWorkers/WorkerGroup.vue [hunk 3]
```diff
@@ -54,15 +63,15 @@ export default {
...
+      return this.machineTypesByCloudProfileName({ cloudProfileName: this.cloudProfileName })
     },
     volumeTypes () {
-      return this.volumeTypesByCloudProfileNameAndZones({ cloudProfileName: this.cloudProfileName })
+      return this.volumeTypesByCloudProfileName({ cloudProfileName: this.cloudProfileName })
```
Comment (petersutter) [inline frontend/src/components/ShootWorkers/WorkerGroup.vue]: volumeTypesByCloudProfileName does this function exist?

File: frontend/src/utils/index.js [hunk 4]
```diff
@@ -348,7 +348,7 @@ export function canLinkToSeed ({ namespace, seedName }) {
   * If we are not in the garden namespace we expect a seed to be present
   * TODO refactor once we have an owner ref on the shoot pointing to the seed
   */
-  return namespace && seedName !== 'garden'
+  return namespace && seedName && seedName !== 'garden'
```
Comment (petersutter) [inline frontend/src/utils/index.js]: fix

File: frontend/src/components/ShootWorkers/WorkerGroup.vue (hunk 3 above)
Comment (grolu) [reply frontend/src/components/ShootWorkers/WorkerGroup.vue]: done - merge error

File: frontend/src/components/ShootWorkers/WorkerGroup.vue (hunk 2 above)
Comment (grolu) [reply frontend/src/components/ShootWorkers/WorkerGroup.vue]: done - merge error

File: frontend/src/components/ShootWorkers/WorkerConfiguration.vue (hunk 1 above)
Comment (grolu) [reply frontend/src/components/ShootWorkers/WorkerConfiguration.vue]: fixed

File: frontend/src/components/ShootWorkers/ManageWorkers.vue [hunk 5]
```diff
@@ -181,12 +184,13 @@ export default {
       this.valid = valid
       this.$emit('valid', this.valid)
     },
-    setWorkersData ({ workers, cloudProfileName, region, zonesNetworkConfiguration, updateOSMaintenance }) {
+    setWorkersData ({ workers, cloudProfileName, region, zonesNetworkConfiguration, zonedCluster, updateOSMaintenance }) {
```
Comment (petersutter) [inline frontend/src/components/ShootWorkers/ManageWorkers.vue]: default: zonedCluster = true

File: frontend/src/components/ShootWorkers/ManageWorkers.vue [hunk 6]
```diff
@@ -181,12 +184,13 @@ export default {
...
       this.cloudProfileName = cloudProfileName
       this.region = region
       this.zonesNetworkConfiguration = zonesNetworkConfiguration
       this.updateOSMaintenance = updateOSMaintenance
       this.setInternalWorkers(workers)
+      this.zonedCluster = zonedCluster !== false
```
Comment (petersutter) [inline frontend/src/components/ShootWorkers/ManageWorkers.vue]: rm

File: frontend/src/store/index.js [hunk 7]
```diff
@@ -161,26 +161,42 @@ const getters = {
...
+  },
+  machineTypesByCloudProfileNameAndRegionAndZones (state, getters) {
+    return ({ cloudProfileName, region, zones }) => {
       const cloudProfile = getters.cloudProfileByName(cloudProfileName)
       if (cloudProfile) {
         const machineTypes = cloudProfile.data.machineTypes
-        const unavailableItems = flatMap(cloudProfile.data.regions, ({ zones }) => {
-          return map(zones, 'unavailableMachineTypes')
+        const regionObject = find(cloudProfile.data.regions, { name: region })
+        const unavailableItems = flatMap(get(regionObject, 'zones', []), zone => {
```
Comment (petersutter) [inline frontend/src/store/index.js]: extract zones to `zonesForRegion` variable
filter, then flatmap

same below (volumeTypesByCloudProfileName)

File: frontend/src/store/index.js [hunk 8]
```diff
@@ -161,26 +161,42 @@ const getters = {
...
+          if (includes(zones, zone.name)) {
+            return zone.unavailableMachineTypes
+          }
         })
         return filter(machineTypes, machineType => machineAndVolumeTypePredicate(machineType, unavailableItems))
```
Comment (petersutter) [inline frontend/src/store/index.js]:
```
const machineAndVolumeTypePredicate = unavailableItems => {
  return item => {
    if (item.usable === false) {
      return false
    }
    if (includes(unavailableItems, item.name)) {
      return false
    }
    return true
  }
}
```

`return filter(machineTypes, machineAndVolumeTypePredicate(unavailableItems))`

File: frontend/src/store/index.js (hunk 7 above)
Comment (grolu) [reply frontend/src/store/index.js]: refactored code for more reuse

**Pull request merged**